src/pykeen/models/nbase.py

```diff
@@ -365,23 +372,25 @@ def __init__(
             random_seed=random_seed,
             predict_with_sigmoid=predict_with_sigmoid,
         )
+        self.interaction = interaction_resolver.make(interaction, pos_kwargs=interaction_kwargs)
         self.entity_representations = _prepare_representation_module_list(
             representations=entity_representations,
             num_embeddings=triples_factory.num_entities,
-            shapes=interaction.entity_shape,
+            shapes=self.interaction.entity_shape,
             label="entity",
-            skip_checks=interaction.tail_entity_shape is not None,
+            skip_checks=self.interaction.tail_entity_shape is not None,
         )
         self.relation_representations = _prepare_representation_module_list(
             representations=relation_representations,
             num_embeddings=triples_factory.num_relations,
-            shapes=interaction.relation_shape,
+            shapes=self.interaction.relation_shape,
             label="relation",
         )
-        self.interaction = interaction_resolver.make(interaction, pos_kwargs=interaction_kwargs)
         # Comment: it is important that the regularizers are stored in a module list, in order to appear in
         # model.modules(). Thereby, we can collect them automatically.
         self.weight_regularizers = nn.ModuleList()
+        # Explicitly call reset_parameters to trigger initialization
+        self.reset_parameters_()
```

Comment:
we're not currently using the functionality of the weight_regularizers in the main branch, but #107 might have examples where we need to add regularizers using `ERModel.append_weight_regularizer()`. Doesn't this happen after `super().__init__()` is called?

Reply:
See example in ConvKB: https://github.com/pykeen/pykeen/blob/79b21871af7f56570678f420a518b64d6eaa66dc/src/pykeen/models/unimodal/conv_kb.py#L73-L114

I'm going to make a new issue for porting ConvKB to the new ERModel that should block this PR

Reply:
This is using the old-style base model? If yes, this should not block this PR.

Reply:
no, it's the new-style one. I think we should upgrade the ConvKB model to new-style then see if this PR still works the way as intended on ConvKB

Reply:
I think TransH is also another example with code after the `super().__init__()` to the ERModel that has not been ported over that we should consider